src/main/java/com/pinterest/secor/common/LogFilePath.java

```diff
@@ -146,8 +146,12 @@ public LogFilePath withPrefix(String prefix) {
 
     public String getLogFileParentDir() {
         ArrayList<String> elements = new ArrayList<String>();
-        elements.add(mPrefix);
-        elements.add(mTopic);
+        if(mPrefix.length() > 0) {
```

Comment:
If you want to be safe, you can do:
   if (mPrefix != null && mPrefix.length()) > 0 ) {

On the coding style side, there should be a space after 'if'